practitioner-service/src/main/java/uk/ac/ox/ndph/mts/practitioner_service/controller/PractitionerController.java

```diff
@@ -16,36 +19,41 @@
 import uk.ac.ox.ndph.mts.practitioner_service.model.Response;
 import uk.ac.ox.ndph.mts.practitioner_service.model.RoleAssignment;
 import uk.ac.ox.ndph.mts.practitioner_service.service.EntityService;
-
-import static org.springframework.http.HttpStatus.CREATED;
-import static org.springframework.http.HttpStatus.OK;
-import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+import uk.ac.ox.ndph.mts.security.authorisation.SecurityContextUtil;
 
 import javax.validation.constraints.NotBlank;
 import javax.validation.constraints.NotNull;
 import java.util.List;
 
+import static org.springframework.http.HttpStatus.CREATED;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
 /**
  * Controller for practitioner endpoint of practitioner-service
  */
 @RestController
+@CrossOrigin(origins = "http://localhost:4200")  // See https://www.baeldung.com/spring-cors  
```

Comment:
which endpoint is 4200? is it just for local work? or is it also when deployed on azure?

Reply:
Yes this is for local work - the Angular front end runs on port 4200 in the dev environment.  This means that when the front end makes an http request to the back end, the browser thinks it is a Cross Origin Request and so makes a pre-flight options request.   This annotation tells Spring to give the correct headers to make the browser happy.

Reply:
Hi, following our conversation, we need to test it on Azure see it works there ...

Reply:
It didn't work on Azure.  I discussed this with Mark and we decided to permit CORS from all origins.  Security issue does not arise given the auth we have in place.  (NB Liza, that other problem with logging into the Angular UI was not to do with injection of the clientid after all - Elad helped me eliminate that. The problem was a bug in the Angular config, which Mark was aware of and has already fixed it. his PR for that fix is blocked by a webdriver deployment issue but he's working on that now. We tested it with a workaround.)

Reply:
I have moved the CORS config into the gateway.  Here it is working on Azure https://sakate477uidev.z33.web.core.windows.net/